### PR TITLE
feat(auth): update GitHub sign in implementation

### PR DESCRIPTION
**Pull request by Lyokone**
## Description
update android GitHub sign in implementation to use underlying platform and not expose secret (#8943)

## Related Issues

#8943 
#8950 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [X] All existing and new tests are passing.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] I signed the [CLA].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_auth/firebase_auth/android/src/main/java/io/flutter/plugins/firebase/auth/FlutterFirebaseAuthPlugin.java [hunk 1]
```diff
@@ -1411,6 +1414,31 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
...
+    TaskCompletionSource<Map<String, Object>> taskCompletionSource = new TaskCompletionSource<>();
+
+    cachedThreadPool.execute(
+        () -> {
+          try {
+            FirebaseAuth firebaseAuth = getAuth(arguments);
+            String newEmail =
+                (String) Objects.requireNonNull(arguments.get(Constants.SIGN_IN_PROVIDER));
+
+            OAuthProvider.Builder provider = OAuthProvider.newBuilder("github.com");
```
[the remark this answers is not in the document]
Comment (Lyokone) [reply packages/firebase_auth/firebase_auth/android/src/main/java/io/flutter/plugins/firebase/auth/FlutterFirebaseAuthPlugin.java]: Should be good now

File: packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart [hunk 2]
```diff
@@ -647,6 +647,12 @@ class FirebaseAuth extends FirebasePluginPlatform {
     return _delegate.verifyPasswordResetCode(code);
   }
 
+  Future<UserCredentialPlatform> signInWithAuthProvider(
```
Comment (nilsreichardt) [inline packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart:650]: All the methods have doc comments in this file. Is this intended that this method has not a doc comment?

File: packages/firebase_auth/firebase_auth_platform_interface/lib/src/platform_interface/platform_interface_firebase_auth.dart [hunk 3]
```diff
@@ -517,6 +518,12 @@ abstract class FirebaseAuthPlatform extends PlatformInterface {
     throw UnimplementedError('signInWithEmailLink() is not implemented');
   }
 
+  Future<UserCredentialPlatform> signInWithAuthProvider(
```
Comment (nilsreichardt) [inline packages/firebase_auth/firebase_auth_platform_interface/lib/src/platform_interface/platform_interface_firebase_auth.dart:521]: dito

File: packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart (hunk 2 above)
Comment (Lyokone) [reply packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart]: Added, thanks :)

**Review by russellwheatley — APPROVED** [review]

**Review by Salakar — APPROVED** [review]

**Pull request merged**